src/analytics/provider/analytics.provider.tsx

```diff
@@ -128,24 +133,25 @@ export function AnalyticsProvider({ children }: AnalyticsProviderProps) {
     }
   }, [site, tradeQueue])
 
-  const trackEcoRouterTradeVolume = (trade: Trade) => {
+  const trackEcoRouterTradeVolume = (trade: Trade, chartOption = ChartOption.OFF) => {
     if (!site || !window.fathom) {
       return console.error('Fathom site not found', { site, fathom: window.fathom })
     }
 
     // Attempt to process the trade volume event
     // If it fails, add it to the queue
     trackers
-      .trackEcoRouterTradeVolume(trade, site)
-      .then(() => debug('processed trade volume event', { trade }))
+      .trackEcoRouterTradeVolume(trade, site, chartOption)
+      .then(() => debug('processed trade volume event', { trade, chartOption }))
       .catch(error => {
-        console.error('Error processing trade', { trade, error })
-        const id = computeItemId(trade)
+        console.error('Error processing trade', { trade, chartOption, error })
+        const id = computeItemId(trade, chartOption)
         setTradeQueue(state => ({
           ...state,
           [id]: {
             id,
             item: trade,
+            chartOption,
             status: ItemStatus.PENDING,
             retries: 0,
           },
```

Comment:
I get that this is an old code but I think it would be much cleaner with `async` / `await`.